cli/src/feature.rs

```diff
@@ -151,6 +158,120 @@ impl fmt::Display for CliFeatures {
+            writeln!(
+                f,
+                "\n{}",
+                style("To activate features the tool and cluster feature sets must match, select a tool version that matches the cluster")
+                    .bold())?;
+        } else {
+            if !self.stake_allowed {
+                write!(
+                    f,
+                    "\n{}",
+                    style("To activate features the stake must be >= 95%")
+                        .bold()
+                        .red()
+                )?;
+            }
+            if !self.rpc_allowed {
+                write!(
+                    f,
+                    "\n{}",
+                    style("To activate features the RPC nodes must be >= 95%")
+                        .bold()
+                        .red()
+                )?;
+            }
+        }
+        writeln!(
+            f,
+            "\n\n{}",
+            style(format!("Tool Feature Set: {}", self.tool_feature_set)).bold()
+        )?;
+        let software_versions_title = "Software Version";
+        let feature_set_title = "Feature Set";
+        let stake_percent_title = "Stake";
+        let rpc_percent_title = "RPC";
+        let max_software_versions_len = max(
+            software_versions_title.len(),
+            self.max_software_versions_len,
+        );
+        let max_feature_set_len = max(feature_set_title.len(), self.max_feature_set_len);
+        let max_stake_percent_len = max(stake_percent_title.len(), self.max_stake_percent_len);
+        let max_rpc_percent_len = max(rpc_percent_title.len(), self.max_rpc_percent_len);
+        writeln!(
+            f,
+            "{}",
+            style(format!(
+                "{1:<0$}  {3:<2$}  {5:<4$}  {7:<6$}",
+                max_software_versions_len,
+                software_versions_title,
+                max_feature_set_len,
+                feature_set_title,
+                max_stake_percent_len,
+                stake_percent_title,
+                max_rpc_percent_len,
+                rpc_percent_title,
+            ))
+            .bold(),
+        )?;
+        for feature_set in &self.feature_sets {
+            writeln!(
+                f,
+                "{1:<0$}  {3:>2$}  {5:>4$} {7:>6$}  {8}",
+                self.max_software_versions_len,
+                feature_set.software_versions,
+                self.max_feature_set_len,
+                feature_set.feature_set,
+                self.max_stake_percent_len,
+                feature_set.stake_percent,
+                self.max_rpc_percent_len,
+                feature_set.rpc_percent,
+                if feature_set.me { "<-- me" } else { "" },
+            )?;
+        }
+        writeln!(f)
+    }
+}
+
+impl QuietDisplay for CliClusterFeatureSets {}
+impl VerboseDisplay for CliClusterFeatureSets {}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CliFeatureSet {
+    software_versions: String,
```

Comment:
I considered making this a Vec of versions, but wasn't sure it would gain a whole lot.

Reply:
Hm yeah not sure who the consumers of this are but I think a vec of version strings would be better for consumers